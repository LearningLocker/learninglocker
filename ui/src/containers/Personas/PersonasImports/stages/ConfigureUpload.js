import React from 'react';
import {
  compose,
  withProps,
  withHandlers,
  withState
} from 'recompose';
import { withModel } from 'ui/utils/hocs';
import { Map, List } from 'immutable';
import { modelsSchemaIdSelector } from 'ui/redux/selectors';
import { connect } from 'react-redux';
import classNames from 'classnames';
import {
  importPersonas
} from 'ui/redux/modules/persona';
import ValidationList from 'ui/components/ValidationList';
import ColumnHeaderEditor from '../columnHeaderEditor';

const schema = 'personasImport';

const disabledState = withState('disabled', 'setDisabled', ({ disabled }) => disabled);

const handlers = withHandlers({
  handleImportPersonas: ({
    model,
    importPersonas: doImportPersonas,
    setDisabled
  }) => () => {
    setDisabled(true);
    doImportPersonas({
      id: model.get('_id')
    });
  }
});


export const ConfigureUploadComponent = ({
  model,
  handleImportPersonas,
  disabled = false,
}) => {
  const errors = model.getIn(['errors', 'messages', 'structure']);
  return (
    <div>
      <div
        className={classNames({
          'form-group': true,
          'has-error': model.getIn(['errors', 'messages', 'structure'], false)
        })}>
        <ColumnHeaderEditor
          csvHeaders={model.get('csvHeaders', new List())}
          structure={model.get('structure', new Map())}
          model={model}
          disabled={disabled} />

        {errors !== undefined &&
          (<span className="help-block">
            <ValidationList errors={errors} />
          </span>)
        }
      </div>
      {!disabled &&
        <div className="form-group">
          <button
            className="btn btn-primary pull-right"
            onClick={handleImportPersonas}>
            Import Personas
            </button>
        </div>
      }
    </div>
  );
};

export default compose(
  withProps(({ model }) => ({
    schema,
    id: model.get('_id')
  })),
  withModel,
  connect(
    (state, { schema: connectSchema, id }) => ({
      model: modelsSchemaIdSelector(connectSchema, id, { deep: true })(state)
    }),
    { importPersonas }
  ),
  disabledState,
  handlers
)(
  ConfigureUploadComponent
  );
