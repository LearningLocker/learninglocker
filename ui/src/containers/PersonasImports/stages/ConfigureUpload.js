import React from 'react';
import {
  compose,
  withProps,
  withHandlers
} from 'recompose';
import { withModel } from 'ui/utils/hocs';
import { Map, List } from 'immutable';
import { modelsSchemaIdSelector } from 'ui/redux/selectors';
import { connect } from 'react-redux';
import classNames from 'classnames';

import ColumnHeaderEditor from 'ui/containers/PersonasImports/columnHeaderEditor';
import {
  importPersonas
} from 'ui/redux/modules/persona';
import ValidationList from 'ui/components/ValidationList';

const schema = 'personasImport';

const handlers = withHandlers({
  handleImportPersonas: ({
    model,
    importPersonas: doImportPersonas
  }) => () => {
    doImportPersonas({
      id: model.get('_id')
    });
  }
});


export const ConfigureUploadComponent = ({
  model,
  handleImportPersonas,
  disabled = false,
}) =>
  (
    <div>
      <div
        className={classNames({
          'form-group': true,
          'has-error': model.getIn(['errors', 'messages', 'structure'], false)
        })}>
        <ColumnHeaderEditor
          csvHeaders={model.get('csvHeaders', new List())}
          structure={model.get('structure', new Map())}
          model={model}
          disabled={disabled} />

        {model.getIn(['errors', 'messages', 'structure'], false) &&
          (<span className="help-block">
            <ValidationList errors={model.getIn(['errors', 'messages', 'structure'])} />
          </span>)
        }
      </div>
      {!disabled &&
        <div className="form-group">
          <button
            className="btn btn-primary pull-right"
            onClick={handleImportPersonas}>
            Import Personas
            </button>
        </div>
      }
    </div>
  );

export default compose(
  withProps(({ model }) => ({
    schema,
    id: model.get('_id')
  })),
  withModel,
  connect(
    (state, { schema: connectSchema, id }) => ({
      model: modelsSchemaIdSelector(connectSchema, id, { deep: true })(state)
    }),
    { importPersonas }
  ),
  handlers
)(
  ConfigureUploadComponent
);
