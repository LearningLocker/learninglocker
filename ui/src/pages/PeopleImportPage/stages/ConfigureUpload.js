import React from 'react';
import {
  compose,
  withProps,
  withHandlers,
  withState
} from 'recompose';
import { connect } from 'react-redux';
import classNames from 'classnames';
import { modelsSchemaIdSelector } from 'ui/redux/selectors';
import { withModel } from 'ui/utils/hocs';
import { importPersonas } from 'ui/redux/modules/persona';
import ValidationList from 'ui/components/ValidationList';
import ColumnHeaderEditor from './ColumnHeaderEditor';

const schema = 'personasImport';

const disabledState = withState('disabled', 'setDisabled', ({ disabled }) => disabled);

const handlers = withHandlers({
  handleImportPersonas: ({
    model,
    importPersonas: doImportPersonas,
    setDisabled
  }) => () => {
    setDisabled(true);
    doImportPersonas({
      id: model.get('_id')
    });
  }
});


export const ConfigureUploadComponent = ({
  model, // personasImports model
  handleImportPersonas,
  disabled = false,
}) => {
  const errors = model.getIn(['errors', 'messages', 'structure']);
  return (
    <div>
      <div
        className={classNames({
          'form-group': true,
          'has-error': model.getIn(['errors', 'messages', 'structure'], false)
        })}>
        <ColumnHeaderEditor
          model={model}
          disabled={disabled} />

        {errors !== undefined &&
          (<span className="help-block">
            <ValidationList errors={errors} />
          </span>)
        }
      </div>
      {!disabled &&
        <div className="form-group">
          <button
            className="btn btn-primary pull-right"
            onClick={handleImportPersonas}>
            Import Personas
            </button>
        </div>
      }
    </div>
  );
};

export default compose(
  withProps(({ model }) => ({
    schema,
    id: model.get('_id')
  })),
  withModel,
  connect(
    (state, { schema: connectSchema, id }) => ({
      model: modelsSchemaIdSelector(connectSchema, id, { deep: true })(state)
    }),
    { importPersonas }
  ),
  disabledState,
  handlers
)(
  ConfigureUploadComponent
  );
