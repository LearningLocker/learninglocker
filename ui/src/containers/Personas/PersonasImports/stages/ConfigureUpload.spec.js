import React from 'react';
import { shallow } from 'enzyme';
import 'jest-enzyme';
import { fromJS } from 'immutable';
import { ConfigureUploadComponent } from './ConfigureUpload';

describe('ConfigureUpload', () => {
  it('should render', () => {
    const rendered = shallow(<ConfigureUploadComponent
      model={fromJS({
        csvHeaders: ['test1', 'test2']
      })} />
    );

    expect(rendered).toMatchSnapshot();
  });

  it('should render with errors', () => {
    const rendered = shallow(<ConfigureUploadComponent
      model={fromJS({
        errors: {
          messages: {
            structure: 'Structure is invalid'
          }
        }
      })} />
    );

    expect(rendered).toMatchSnapshot();
  });

  it('should render with disabled', () => {
    const rendered = shallow(<ConfigureUploadComponent
      model={fromJS({
        csvHeaders: ['test1', 'test2']
      })}
      disabled />
    );

    expect(rendered).toMatchSnapshot();
  });
});
