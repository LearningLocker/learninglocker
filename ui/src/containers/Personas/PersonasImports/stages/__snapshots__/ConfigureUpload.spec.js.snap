// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureUpload should render 1`] = `
<div>
  <div
    className="form-group"
  >
    <WithStyles(ColumnHeaderEditorComponent)
      csvHeaders={
        Array [
          "test1",
          "test2",
        ]
      }
      disabled={false}
      model={
        Object {
          "csvHeaders": Array [
            "test1",
            "test2",
          ],
        }
      }
      structure={Object {}}
    />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-primary pull-right"
    >
      Import Personas
    </button>
  </div>
</div>
`;

exports[`ConfigureUpload should render with disabled 1`] = `
<div>
  <div
    className="form-group"
  >
    <WithStyles(ColumnHeaderEditorComponent)
      csvHeaders={
        Array [
          "test1",
          "test2",
        ]
      }
      disabled="true"
      model={
        Object {
          "csvHeaders": Array [
            "test1",
            "test2",
          ],
        }
      }
      structure={Object {}}
    />
  </div>
</div>
`;

exports[`ConfigureUpload should render with errors 1`] = `
<div>
  <div
    className="form-group has-error"
  >
    <WithStyles(ColumnHeaderEditorComponent)
      csvHeaders={Array []}
      disabled={false}
      model={
        Object {
          "errors": Object {
            "messages": Object {
              "structure": "Structure is invalid",
            },
          },
        }
      }
      structure={Object {}}
    />
    <span
      className="help-block"
    >
      <WithStyles(ValidationList)
        errors="Structure is invalid"
      />
    </span>
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-primary pull-right"
    >
      Import Personas
    </button>
  </div>
</div>
`;
